packages/opentelemetry-sdk-node/src/sdk.ts

```diff
@@ -0,0 +1,165 @@
+   */
+  public constructor(configuration: Partial<NodeSDKConfiguration> = {}) {
+    this._resource = configuration.resource ?? new Resource({});
+
+    this._autoDetectResources = configuration.autoDetectResources ?? true;
+
+    if (configuration.spanProcessor || configuration.traceExporter) {
+      const tracerProviderConfig = {} as NodeTracerConfig;
+
+      if (configuration.defaultAttributes) {
+        tracerProviderConfig.defaultAttributes =
+          configuration.defaultAttributes;
+      }
+      if (typeof configuration.logLevel === 'number') {
+        tracerProviderConfig.logLevel = configuration.logLevel;
+      }
+      if (configuration.logger) {
+        tracerProviderConfig.logger = configuration.logger;
+      }
+      if (configuration.plugins) {
+        tracerProviderConfig.plugins = configuration.plugins;
+      }
+      if (configuration.sampler) {
+        tracerProviderConfig.sampler = configuration.sampler;
+      }
+      if (configuration.traceParams) {
+        tracerProviderConfig.traceParams = configuration.traceParams;
+      }
+
+      const spanProcessor =
+        configuration.spanProcessor ??
+        new BatchSpanProcessor(configuration.traceExporter!);
+
+      this.configureTracerProvider(
+        tracerProviderConfig,
+        spanProcessor,
+        configuration.contextManager,
+        configuration.httpTextPropagator
+      );
+    }
+
+    if (configuration.metricExporter) {
+      const meterConfig: MeterConfig = {};
+
+      if (configuration.metricBatcher) {
+        meterConfig.batcher = configuration.metricBatcher;
+      }
+      if (configuration.metricExporter) {
+        meterConfig.exporter = configuration.metricExporter;
+      }
+      if (typeof configuration.metricInterval === 'number') {
+        meterConfig.interval = configuration.metricInterval;
+      }
+      if (typeof configuration.logLevel === 'number') {
+        meterConfig.logLevel = configuration.logLevel;
+      }
+      if (configuration.logger) {
+        meterConfig.logger = configuration.logger;
+      }
+
+      this.configureMeterProvider(meterConfig);
+    }
+  }
+
+  /** Set configurations required to register a NodeTracerProvider */
+  public configureTracerProvider(
+    tracerConfig: NodeTracerConfig,
+    spanProcessor: SpanProcessor,
+    contextManager?: ContextManager,
+    httpTextPropagator?: HttpTextPropagator
+  ) {
+    this._tracerProviderConfig = {
+      tracerConfig,
+      spanProcessor,
+      contextManager,
+      httpTextPropagator,
+    };
+  }
+
+  /** Set configurations needed to register a MeterProvider */
+  public configureMeterProvider(config: MeterConfig) {
+    this._meterProviderConfig = config;
+  }
+
+  /** Detect resource attributes from the execution environment */
+  public async detectResources() {
+    this.addResource(await detectResources());
+  }
+
+  /** Manually add a resource */
+  public addResource(resource: Resource) {
+    this._resource.merge(resource);
```

Comment:
I found the 🐛 I mentioned during the last SIG meeting. `merge` does not mutate the resource.

```suggestion
    this._resource = this._resource.merge(resource);
```